### DISCUSSION TITLE
_replicator document with parameter {"two_way": true} for two-way continuous replication

### DISCUSSION THREAD

**Discussion by SinanGabel** [apache/couchdb#4206] [Ideas] [answered]
Example of a _replicator document with two way continuous replication between the source and target.

```

{
  "_id": "some id",
  "_rev": "some rev",
  "source": {
    "headers": {
      "Authorization": "Basic some auth"
    },
    "url": "http://source url"
  },
  "target": {
    "headers": {
      "Authorization": "Basic some auth"
    },
    "url": "http://target url"
  },
  "create_target": true,
  "continuous": true,
  "owner": "some owner",
  "two_way": true
}

```

**Comment by nickva** [comment, marked as answer]
Thanks. I assume this is a suggestion for a new feature?

It's an interesting idea,  though in that case `source` and `target` don't quite make sense. I guess they could just be `"endpoints" : [...]` that is an list of endpoints between which we want to replicate, (could be 2 or 3, for example). And, maybe, another parameter like `"topology"  : "ring" | "bidirectional_ring" | "fully_connected"`. Then ,we'd end with a replication job group which monitors other child jobs or something of that sort.

**Reply by SinanGabel** [reply]
Hi, 

Thanks, yes, it is a new feature idea, a kind of "convenience" method: 

I was thinking that instead of having to add two _replicator documents - doing {source A => target B} and {source B => target A}  - one could use a "two_way: true" setting such that it would automatically do - source A <=> target B two-way replication. 

True, in some cases source and target would not make sense, but one could maybe interpret the source as a kind of "master";
for example, in the case where there is already data in both the source A and target B databases.
